Tests/SwiftSyntaxTest/TriviaTests.swift

```diff
@@ -69,4 +69,325 @@ class TriviaTests: XCTestCase {
+      ["Some block comment spread on many lines"]
+    )
+
+    XCTAssertEqual(
+      Trivia(pieces: [
+        .blockComment(
+          """
+          /*
+          *  Some block comment
+          *  spread on many lines
+          */
+          """
+        )
+      ]).commentValues,
+      ["Some block comment spread on many lines"]
+    )
+
+    // MARK: doc block comment
+
+    XCTAssertEqual(
+      Trivia(pieces: [.docBlockComment("")]).commentValues,
+      [""]
+    )
+
+    XCTAssertEqual(
+      Trivia(pieces: [.docBlockComment("Some doc block comment")]).commentValues,
+      ["Some doc block comment"]
+    )
+
+    XCTAssertEqual(
+      Trivia(pieces: [.docBlockComment("/** Some doc block comment */")]).commentValues,
+      ["Some doc block comment"]
+    )
+
+    XCTAssertEqual(
+      Trivia(pieces: [
+        .docBlockComment("/** Some doc block comment"),
+        .docBlockComment("* spread on many lines */"),
+      ]).commentValues,
+      [
+        "Some doc block comment spread on many lines"
+      ]
+    )
+
+    XCTAssertEqual(
+      Trivia(pieces: [
+        .docBlockComment("/** Some doc block comment"),
+        .docBlockComment("* spread on many lines"),
+        .docBlockComment("*/"),
+      ]).commentValues,
+      [
+        "Some doc block comment spread on many lines"
+      ]
+    )
+
+    XCTAssertEqual(
+      Trivia(pieces: [
+        .docBlockComment("/** Some doc block comment"),
+        .docBlockComment("* spread on many lines */"),
+        .docBlockComment("/** Another doc block comment */"),
+      ]).commentValues,
+      [
+        "Some doc block comment spread on many lines",
+        "Another doc block comment",
+      ]
+    )
+
+    XCTAssertEqual(
+      Trivia(pieces: [
+        .docBlockComment("/** Some doc block comment"),
+        .docBlockComment("* spread on many lines */"),
+        .newlines(2),
+        .docBlockComment("/** Another doc block comment */"),
+      ]).commentValues,
+      [
+        "Some doc block comment spread on many lines",
+        "Another doc block comment",
+      ]
+    )
+
+    XCTAssertEqual(
+      Trivia(pieces: [
+        .docBlockComment(
+          """
+          /**
+          Some doc block comment
+          spread on many lines
+          */
+          """
+        )
+      ]).commentValues,
+      ["Some doc block comment spread on many lines"]
+    )
+
+    XCTAssertEqual(
+      Trivia(pieces: [
+        .docBlockComment(
+          """
+          /**
+          *  Some doc block comment
+          *  spread on many lines
+          */
+          """
+        )
+      ]).commentValues,
+      ["Some doc block comment spread on many lines"]
+    )
+
+    // MARK: Mixing comment styles
+
+    XCTAssertEqual(
+      Trivia(pieces: [
+        .docBlockComment(
+          """
+          /**
+          *  Some doc block comment
+          *  // spread on many lines
+          *  with a line comment
+          */
+          """
```

Comment:
I think it’s quite common for the `*` to be indented by a single space to align with the first `*` of `/*`. Could you also add a test case for something like

```swift
          /**
           *  Some doc block comment
           *  with a line comment
          */
          """
```